syntax = "proto3";

package rpc.v1;

import "rpc/v1/ledger_objects.proto";

// A request to get info about an account.
message GetAccountInfoRequest {
    // The address to get info about.
    string address = 1;

    bool strict = 2;

    oneof ledger_index {
        string ledger_index_shortcut_string = 3;
        uint32 ledger_index_seq = 4;
        bytes ledger_hash = 5;
    }

    bool queue = 6;

    bool signer_lists = 7;
}

// Response to GetAccountInfo RPC
message GetAccountInfoResponse {

    AccountRoot account_data = 1;

    SignerList signer_list = 2;

    uint32 ledger_index = 3;
    
    QueueData queue_data = 4;

    bool validated = 5; 
}

// Aggregate data about queued transactions
message QueueData {

    uint32 txn_count = 1;

    bool auth_change_queued = 2;

    uint32 lowest_sequence = 3;

    uint32 highest_sequence = 4;

    uint64 max_spend_drops_total = 5;

    repeated QueuedTransaction transaction = 6;
}

// Data about a single queued transaction
message QueuedTransaction {
    bool auth_change = 1;

    uint64 fee = 2;

    uint64 fee_level = 3;

    uint64 max_spend_drops = 4;

    uint32 seq = 5;

    uint32 last_ledger_sequence = 6;
}
